### PR TITLE
fix(mac): avoid crash on startup with macOS 10.15 (Catalina)

### PR DESCRIPTION
**Pull request by sgschantz**
To read the state and gain access to the Accessibility permission under the macOS Privacy & Security settings, Keyman calls the API `CGPreflightPostEventAccess`.  This API is [documented as being available with with macOS 10.15](https://developer.apple.com/documentation/coregraphics/3656520-cgpreflightposteventaccess?language=objc) (Catalina), but, based on experience, this does not appear to be the case. Limit use of API `CGPreflightPostEventAccess` to macOS 11.0 (Big Sur) and later, and, instead, use an older API, `AXIsProcessTrusted`, with OS versions before 11.0.

Fixes: #12295

# User Testing

## TEST_RUNS_SUCCESSFULLY_ON_MACOS_CATALINA

1. Install this version of Keyman on a system running macOS Catalina
1. Confirm that keyboards can be installed and that they appear in the Keyman menu
1. Type with the installed Keyman keyboards and confirm that keystrokes are processed as expected

### CODE REVIEW

**Comment by dinakaranr** [comment]
# Test Results
I tested this issue with the attached "Keyman 18.0.104-alpha-local" build on the MacOS Catalina. Here is my observation.

* **TEST_RUNS_SUCCESSFULLY_ON_MACOS_CATALINA (Passed):**
1. Installed the Keyman keyboard package version 18.0.104.dmg
2. Open the configuration dialog.
3. Installed the "Khmer" & "Tamil99" keyboards. (e.g. IPA SIL)
4. Open the "Text Editor" application.
5. Select the Khmer keyboard in the toolbar.
6. Type some texts and then the text reflects it.
It works well. Thank you.

**Comment by sgschantz** [comment]
@mcdurdin will cherry-pick to stable 17.0 after review

File: mac/Keyman4MacIM/Keyman4MacIM/Privacy/PrivacyConsent.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/**
+/*
```
Comment (rc-swag) [inline mac/Keyman4MacIM/Keyman4MacIM/Privacy/PrivacyConsent.m:1]: was it deliberate to remove the /**

**Review by rc-swag — APPROVED** [review]
lgtm

File: mac/Keyman4MacIM/Keyman4MacIM/Privacy/PrivacyConsent.m (hunk 1 above)
Comment (sgschantz) [reply mac/Keyman4MacIM/Keyman4MacIM/Privacy/PrivacyConsent.m:1]: Yes, @mcdurdin said that we should avoid java-doc style comments for the header but elsewhere in the file is ok.

Comment (mcdurdin) [reply mac/Keyman4MacIM/Keyman4MacIM/Privacy/PrivacyConsent.m:1]: Yep, this is in the style guide.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.105-alpha](https://keyman.com/downloads/releases/alpha/18.0.105)

**Pull request merged**